### PR TITLE
docs: Add Node.js SDK docs

### LINKED ISSUE #3640: NodeJS Documentation
### What is the issue?

We will need the following for the NodeJS launch.
- New section in docs
- Get Started Guide
- API reference guide
- Installation
- FAQ

@slumbering  can help support as needed since he is NodeJS SDK release lead.

### PR DESCRIPTION
**Pull request by vikram-dagger**
This commit contains the documentation for the Node.js SDK.

Closes #3640

### CODE REVIEW

**Comment by vikram-dagger** [comment]
@aluzzardi should we add a linter for the typescript code snippets?

File: docs/current/sdk/nodejs/783645-get-started.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+## Step 4: Test against multiple Node.js versions
+
+Now that the Node.js CI tool can test the application against a single Node.js version, the next step is to extend it for multiple Node.js versions.
+
+Replace the `build.ts` file from the previous step with the version below (highlighted lines indicate changes):
+
+```typescript file=snippets/get-started/step4/build.ts
+```
+
+This revision of the CI tool does much the same as before, except that it now supports multiple Node.js versions.
```
Comment (mircubed) [inline docs/current/sdk/nodejs/783645-get-started.md:136]: This sentence seems off to me. Maybe change to "This revision of the CI tool is the same as before, except that it now supports multiple Node.js versions.

**Comment by mircubed** [comment]
not sure if we can fix this, but it looks like a line spacing issue here? It isn't a dealbreaker for getting the docs out though. 
Screen Shot 2022-11-15 at 8 20 52 AM

File: docs/current/sdk/nodejs/783645-get-started.md (hunk 1 above)
Comment (vikram-dagger) [reply docs/current/sdk/nodejs/783645-get-started.md]: Modified

**Comment by vikram-dagger** [comment]
> not sure if we can fix this, but it looks like a line spacing issue here? It isn't a dealbreaker for getting the docs out though. Screen Shot 2022-11-15 at 8 20 52 AM

I think this might be related to our theme CSS. Not sure who knows about this though - maybe @jpadams can comment?

**Review by gerhard — APPROVED** [review]
Looks good to me 👍

Merge PR via `Create merge commit` so that we can revert https://github.com/dagger/dagger/pull/3823/commits/d0b46f3473f8d5c6eac6e1286c6dbbd9276cb996

**Pull request merged**